### ISSUE TITLE
Proper type parsing of C++ pointer to arrays

### ISSUE THREAD

**Issue by oxisto** [Fraunhofer-AISEC/cpg#731]
Pointer to an array of 4 chars: `char (*a)[4]`, which should have the type of `char (*)[4]`
Array of 4 pointers to a char: `char* b[4]`, which should have the type of `char *[4]`

**Comment by KuechA** [comment]
The CDT C++ parser cannot parse this instruction correctly. Indeed, it seems to be valid C-syntax but it seems to be invalid in C++. We have only one language frontend for both cases (C and C++) and only use CDT's C++ parser, which seems to be the cause of this bug.

**Comment by oxisto** [comment]
> The CDT C++ parser cannot parse this instruction correctly. Indeed, it seems to be valid C-syntax but it seems to be invalid in C++. We have only one language frontend for both cases (C and C++) and only use CDT's C++ parser, which seems to be the cause of this bug.

Are you sure? `clang++` parses this without problem

**Comment by KuechA** [comment]
You're right, the type is valid in C++ and should be represented as `PointerType[PointerType[name="char[]"]]` in the CPG but is a `PointerType[name="char[]"]`.

I got confused with our initial code snippet. When initializing the array `char (*a)[4] = "foo";` (which is ok in C but not in C++), this leads to an additional problem where the name of the variable cannot be identified.
